src/flint/types/fmpq_mpoly.pxd

```diff
@@ -0,0 +1,36 @@
+from flint.flint_base.flint_base cimport flint_mpoly
+from flint.flint_base.flint_base cimport flint_mpoly_context
+
+from flint.flintlib.fmpq_mpoly cimport *
+from flint.flintlib.flint cimport slong
+
+from flint.types.fmpz_mpoly cimport fmpz_mpoly_ctx
+
+cdef inline init_fmpq_mpoly(fmpq_mpoly var, fmpq_mpoly_ctx ctx):
+    var.ctx = ctx
+    fmpq_mpoly_init(var.val, ctx.val)
+    var._init = True
+
+cdef inline create_fmpq_mpoly(fmpq_mpoly_ctx ctx):
+    cdef fmpq_mpoly var
+    var = fmpq_mpoly.__new__(fmpq_mpoly)
+    var.ctx = ctx
+    fmpq_mpoly_init(var.val, ctx.val)
+    var._init = True
+    return var
```

Comment:
Setting `_init` from outside like this seems suspicious...

Reply:
This is needed because we are using `__new__` to bypass the `__init__` function. `_init` is needed to prevent a potential double free of the flint objects in `__dealloc__`, which Cython doesn't guarantee is only called once.

I've also just checked the Cython code base to see what they do during `__dealloc__` and they also guard  their deallocations
https://github.com/search?q=repo%3Acython%2Fcython%20__dealloc__&type=code